ethminer/MinerAux.h

```diff
@@ -272,6 +302,47 @@ class MinerCLI
 		{
 			m_exit = true;
 		}
+		else if ((arg == "-P") && (i + 1 < argc))
+		{
+			string url = argv[++i];
+			URI uri;
+			try {
+				uri = url;
+			}
+			catch (...) {
+				cerr << "Bad endpoint address: " << url << endl;
+				BOOST_THROW_EXCEPTION(BadArgument());
+			}
+			if (!uri.KnownScheme())
+			{
+				cerr << "Unknown URI scheme " << uri.Scheme() << endl;
+				BOOST_THROW_EXCEPTION(BadArgument());
+			}
+			if (m_ep_ix >= k_max_endpoints)
+			{
+				cerr << "Too many endpoints. Maximum is " << k_max_endpoints << endl;
+				BOOST_THROW_EXCEPTION(BadArgument());
+			}
+			m_endpoints[m_ep_ix] = PoolConnection(uri.Host(), uri.Port(), uri.User(), uri.Pswd(), uri.ProtoSecureLevel(), uri.ProtoVersion());
```

Comment:
For now this is fine, but i guess we should use our URI in PoolConnection directly.

Reply:
Yes, that's intended. Getting there.

Reply:
Actually, no. For now we need to keep the indiviual attributes in PoolConnection since with legacy parameters we write them individually. URI attributes are read only. Once legacy support is dropped, this would be recommended.